### PR TITLE
Yaml Enhancements

### LINKED ISSUE #3306: yaml enhancements
- [x] yaml: specify coarse/fine resolution for all parameters explicitly

Currently, some bipolar ScaleConverters for Modulation Amounts do some extra work (multiply by 2) to provide the intended Coarse/Fine denominators.

We should leave this up to the Yaml definitions and set Coarse/Fine resolutions there explicitly. Then, ScaleConverters wouldn't need any extra rules as everything is predefined.

It is conceivable to refactor the parameter resolutions as ScaleType properties, which would simplify the parameter definition.

Example:
modulateable Parameter with BIPOLAR_100_PERCENT (-1000 ... 1000 fine steps)
--> mod amount with BIPOLAR_200_PERCENT (-2000 ... 2000 fine steps - instead of former -1000 ... 1000 steps)

The step sizes of Mod Amounts should reflect step sizes of associated Control Positions.

- [x] process: also copy the generated overview.html (helping developers, f.e. finding the next free parameter id)

- corrections
 - [x] Cabinet: Hi/Lo Cut short Labels (missing "Hi"/"lo")
 - [x] Output Mixer: Mod Amount Unit should be dB, not %

### LINKED ISSUE #3323: Parameter Info: mention sync behavior for corresponding parameters
> Finde ich gut. Ich würde diesen Satz noch ans Ende setzen:
> 
> `The value of the loop parameter is sampled only with every key down and key up event.`
_Originally posted by @davidm2894 in https://github.com/nonlinear-labs-dev/C15/issues/3158#issuecomment-1268433337_

Latest Proposal
`Changing this parameter is only effective when a key is pressed(/released).`

This applies for:
~~~
Env_A_Lvl_Vel,
Env_A_Att_Vel,
Env_A_Rel_Vel,
Env_A_Lvl_KT,
Env_A_Time_KT,
Env_B_Lvl_Vel,
Env_B_Att_Vel,
Env_B_Rel_Vel,
Env_B_Lvl_KT,
Env_B_Time_KT,
Env_C_Lvl_Vel,
Env_C_Att_Vel,
Env_C_Rel_Vel,
Env_C_Lvl_KT,
Env_C_Time_KT,
Env_A_Att_Curve,
Env_B_Att_Curve,
Env_C_Att_Curve,
Env_A_Dec_1_Vel,
Env_A_Dec_2_Vel,
Env_B_Dec_1_Vel,
Env_B_Dec_2_Vel,
Env_C_Retr_H,
Osc_A_Reset,
Osc_B_Reset,
Voice_Grp_Fade_From,
Voice_Grp_Fade_Range,
Env_C_Dec_1_Vel,
Env_C_Dec_2_Vel,
Env_C_Loop,
Split_Point
~~~

### PR DESCRIPTION
**Pull request by MatthiasSeeber**
- [x] made Descriptors constexpr
- [x] removed nltools dependency in configuration
- [x] introduced Parameter Groups
- [x] preparation for #2995: added polyphonic and monophonic Parameter Types
- [x] provide group elements (PIDs) for each group
- [x] provide display value range (min, max) for scale types
- [x] closes #3323: update parameter infos, properly classify `Poly_Sync::Split_Point`
- [x] fixed yaml inactive bug

closes #3306:
- [x] added overview.html to generated output
- [x] corrected Cabinet and Output Mixer Definitions
- [x] refactor coarse/fine definitions (per parameter, not per scale type)

finally:
- [x] resolve merge conflicts
- [x] cmake: compile Configuration project before nltools (allowing for dependencies)

### CODE REVIEW

File: projects/epc/playground/src/parameters/ScaleParameter.cpp [hunk 1]
```diff
@@ -8,7 +8,7 @@
 #include "scale-converters/dimension/NoteDimension.h"
 #include "parameter_declarations.h"
 #include "parameters/scale-converters/Fine12STScaleConverter.h"
```
Comment (MatthiasSeeber) [inline projects/epc/playground/src/parameters/ScaleParameter.cpp:10]: See above, no apparent reason to include ScaleConverters?

File: projects/epc/playground/src/parameters/scale-converters/FineBipolar72StScaleConverter.h [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+#include "LinearScaleConverter.h"
+
+class FineBipolar72StScaleConverter : public LinearScaleConverter
+{
+ public:
+  FineBipolar72StScaleConverter();
+  ~FineBipolar72StScaleConverter() override = default;
```
Comment (MatthiasSeeber) [inline projects/epc/playground/src/parameters/scale-converters/FineBipolar72StScaleConverter.h:9]: This deviates from the standard overrides used in ScaleConverters - any reason?
(also see FineBipolar200StScaleConverter)

**Review by MatthiasSeeber — COMMENTED** [review]
At this point, only one final task remains:
- [x] modify cmake to compile `shared/configuration` before `shared/nltools`.

With reworked ScaleConverters, all parameters with a fixed number of decimal (post comma) digits now behave as intended:
- [x] each coarse or fine step leads to a distinct display value (no more value displays not changing despite incrementing the position)
- [x] modulation amounts now reflect control position resolutions. The step sizes for both aspects correspond.

However, parabolic times (Flanger Time, Echo Time) use a rounding mechanism with a variable amount of decimal digits.
- [ ] Here, the display value can still remain unchanged when stepping through the control position with fine increments.

---

Here are the outputs of `playground-test` and `acceptance-tests`, they were run with [commit 98e7ec3](https://github.com/nonlinear-labs-dev/C15/pull/3309/commits/98e7ec33c5e40adfbc791e4bbaa86159c35e02f0) - before the merge of `milestone-maintenance`. Besides lots of error messages and some fails I couldn't identify any failing test due to changes on this branch.

[acceptance-tests.txt](https://github.com/nonlinear-labs-dev/C15/files/9689886/acceptance-tests.txt)
[playground-test.txt](https://github.com/nonlinear-labs-dev/C15/files/9689887/playground-test.txt)

**Comment by nonlinear-labs** [comment]
Bei Oscillator A/B - Pitch - Env C ist der MC Amount noch in Schritten von 2 st und 0.02 st quantisiert.

**Comment by MatthiasSeeber** [comment]
@nonlinear-labs 

> Bei Oscillator A/B - Pitch - Env C ist der MC Amount noch in Schritten von 2 st und 0.02 st quantisiert.

Ja. Es gibt noch noch kein update.tar von diesem Pull Request hier, alle betroffenen Parameter (etliche) sind in [commit 9980af0](https://github.com/nonlinear-labs-dev/C15/pull/3309/commits/9980af0b93846f61563488a29ae028c9b7ab397e) einsehbar. Getestet hatte ich bereits. Ich habe eben auch nochmal im Yaml nachgeschaut, die Pitch Env C Mod Amounts sind korrekt.

**Comment by MatthiasSeeber** [comment]
@schlenkibus
I could not find any oddities with yaml-nltools either (except for maybe tests?). Anyway, please notice the code comments in my review above. Parabolic Time issues will be addressed in a ticket soon. Apart from this, we should be good to go.

**Review by schlenkibus — APPROVED** [review]

**Pull request merged**